### PR TITLE
feat(api): Recommendations v1 Service

### PR DESCRIPTION
**Pull request by dmurray-lacework**
## Summary
Add support for v1 Recommendations api `v1/external/recommendations/` to go-sdk

## How did you test this change?
Run new examples `api/_examples/recommendations/main.go`
Unit test `api/recommendations_test.go`

## Issue
https://lacework.atlassian.net/browse/ALLY-933

### CODE REVIEW

File: api/recommendations.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+func (svc *RecommendationsService) AwsList() ([]Recommendation, error) {
+	return svc.list(AwsRecommendation)
+}
+
+func (svc *RecommendationsService) AzureList() ([]Recommendation, error) {
+	return svc.list(AzureRecommendation)
+}
+
+func (svc *RecommendationsService) GcpList() ([]Recommendation, error) {
```
Comment (afiune) [inline api/recommendations.go:45]: There are two operation `List()` and `Patch()` for three clouds `Aws`, `Gcp`, and `Azure`. I think it makes
more sense to nest these as separate services inside Recommendations.
```
apiClient.Recommendations.Aws.List()
apiClient.Recommendations.Aws.Patch()
apiClient.Recommendations.Gcp.List()
apiClient.Recommendations.Gcp.Patch()
apiClient.Recommendations.Azure.List()
apiClient.Recommendations.Azure.Patch()

File: api/recommendations.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	response RecommendationResponse,
+	err error,
+) {
+	err = svc.client.RequestEncoderDecoder("PATCH", fmt.Sprintf(apiRecommendations, cloudType), recommendations, &response)
+	return
+}
+
+type RecommendationState map[string]string
+
+type RecommendationData map[string]RecommendationEnabled
```
Comment (afiune) [inline api/recommendations.go:77]: Could we reserve these namespaces by adding "v1" to their names.

File: api/v1_recommendations_test.go [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+        }
+    ],
+    "ok": true,
+    "message": "SUCCESS"
+}`
+}
+
+func ListRecommendations() string {
+	return `{
+    "data": [
```
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply api/v1_recommendations_test.go:120]: It's quite large, it's the full response but a little unnecessary overhead to read in JSON files

[the remark this answers is not in the document]
Comment (rmoles) [reply api/v1_recommendations_test.go:120]: 🤷 happy enough to leave it as is. Just a personal preference to put large response objects into a JSON file.

**Review by rmoles — APPROVED** [review]
LGTM

**Pull request merged**